src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,178 @@
+import {
+    OrderActionCreator, OrderRequestBody } from '../../../order';
+
+import { GooglePayPaymentInitializeOptions, GooglePayPaymentProcessor } from './';
+import {
+    GooglePaymentData,
+    PaymentMethodData,
+} from './googlepay';
+
+export default class GooglePayPaymentStrategy extends PaymentStrategy {
+    private _googlePayOptions?: GooglePayPaymentInitializeOptions;
+    private _methodId?: string;
+    private _walletButton?: HTMLElement;
+
+    constructor(
+        store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paymentStrategyActionCreator: PaymentStrategyActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _googlePayPaymentProcessor: GooglePayPaymentProcessor
+    ) {
+        super(store);
+    }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        this._methodId = options.methodId;
+
+        this._googlePayOptions = options.googlepay;
+
+        return this._googlePayPaymentProcessor.initialize(this._methodId)
+            .then(() => {
+                if (!options.googlepay) {
+                    throw new InvalidArgumentError('Unable to initialize payment because "options.googlepay" argument is not provided.');
+                }
+
+                const walletButton = options.googlepay.walletButton && document.getElementById(options.googlepay.walletButton);
+
+                if (walletButton) {
+                    this._walletButton = walletButton;
+                    this._walletButton.addEventListener('click', this._handleWalletButtonClick);
+                }
+
+                return super.initialize(options);
+            });
+    }
+
+    deinitialize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (this._walletButton) {
+            this._walletButton.removeEventListener('click', this._handleWalletButtonClick);
+        }
+
+        this._walletButton = undefined;
+
+        return Promise.all([
+            this._googlePayPaymentProcessor.deinitialize(),
+        ]).then(() => super.deinitialize(options));
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this._googlePayOptions) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.googlepay" argument is not provided.');
+        }
+
+        const {
+            onError = () => {},
+            onPaymentSelect = () => {},
+        } = this._googlePayOptions;
+
+        return Promise.resolve(this._getPayment())
+            .then(payment => {
+                if (!payment.paymentData.nonce || !payment.paymentData.cardInformation) {
+                    return this._googlePayPaymentProcessor.displayWallet()
```

Comment:
This block of code seems to be duplicated from `_handleWalletButtonClick`. We need to find a way to keep it DRY. It's not critical, so we could just leave a comment here `TODO: Find a way to share the code with _handleWalletButtonClick 
 method`.